urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/gui/OrderEntryDisplay.java

```diff
@@ -0,0 +1,292 @@
+
+        jPanel1Layout.linkSize(javax.swing.SwingConstants.HORIZONTAL, new java.awt.Component[]{orderPrice, orderVolume});
+
+        jPanel1Layout.setVerticalGroup(
+                jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+                .addComponent(jSeparator1)
+                .addGroup(jPanel1Layout.createSequentialGroup()
+                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.CENTER)
+                .addComponent(jLabel1)
+                .addComponent(orderPrice, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
+                .addComponent(jButton1)
+                .addComponent(mktCheck)
+                .addComponent(buyRadio))
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.CENTER)
+                .addComponent(jButton2)
+                .addComponent(orderVolume, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
+                .addComponent(jLabel2)
+                .addComponent(sellRadio))
+                .addGap(0, 0, Short.MAX_VALUE))
+                .addComponent(jSeparator2));
+
+        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
+        getContentPane().setLayout(layout);
+        layout.setHorizontalGroup(
+                layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+                .addComponent(jPanel1, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE));
+        layout.setVerticalGroup(
+                layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+                .addComponent(jPanel1, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE));
+
+        pack();
+    }// </editor-fold>//GEN-END:initComponents
+
+    /**
+     * Public method sets the visible of display.
+     *
+     * @param p
+     */
+    public void setVisible(String p) {
+        product = p;
+        orderPrice.setText("");
+        orderVolume.setText("");
+        buyRadio.setSelected(false);
+        sellRadio.setSelected(false);
+
+        setTitle("Order Entry for " + product);
+        super.setVisible(true);
+
+    }
+
+    private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton1ActionPerformed
+
+        try {
+            if (orderPrice.getText().trim().isEmpty() && !mktCheck.isSelected()) {
+                throw new GUIDataException("Buy price cannot be empty if price is not MKT");
+            }
+
+            if (orderVolume.getText().trim().isEmpty()) {
+                throw new GUIDataException("Buy Volume cannot be empty");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [GUIDataException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [231].  Consider adding a catch clause of this specific exception in the outer try statement.